cranelift-codegen/src/binemit/mod.rs

```diff
@@ -63,6 +66,35 @@ impl fmt::Display for Reloc {
     }
 }
 
+/// Container for offsets in a vector of compiled code and its supporting read-only data.
+///
+/// The code starts at offset 0 and is followed optionally by relocatable jump tables and copyable
+/// (raw binary) read-only data.  Any padding between sections is always part of the section that
+/// precedes the boundary between the sections.
+///
+/// The offsets to the jumptables and rodata always point past the code, even if there are no tables
+/// and rodata.
+#[derive(PartialEq)]
+pub struct CodeOffsets {
+    /// Number of bytes of machine code (the code starts at offset 0).
+    pub code_size: CodeOffset,
+
+    /// Offset of any relocatable jump tables, or equal to rodata if there are no jump tabes.
+    pub jumptables: CodeOffset,
+
+    /// Number of bytes of jumptables.
+    pub jumptables_size: CodeOffset,
```

Comment:
Here and below (for `rodata_size`): it seems this information duplicates what's provided by `jumptables`, since `jumptables == code_size + jumptables_size`. Maybe we could get rid of one of these two?

If we don't want to get rid of them, it seems weird that a CodeOffset may represent a size; could we just use an `usize` there?

Reply:
I could get rid of eg jumptables and rodata since their offsets can be computed from the sizes, but I'd want to introduce methods to compute them because it's unreasonable for clients to have to perform those computations themselves.  I can investigate that.

Re usize vs CodeOffset, I thought about this but was not sure what to do here.  CodeOffset is perhaps the "wrong" type because it's an offset, but it's the "right" type because no size can be greater than the CodeOffset's range, while usize does allow for a size greater than that.

I think the problem here is the word "offset" in that type name.  This is not actually a type that represents an offset abstractly; it is for sure an unsigned integer type that represents positions in a code segment and as such can (and should) be used to represent the length of the segment as well (the position after the last byte).

With that background, I think I will continue to use the CodeOffset to represent the length.

Reply:
Created getters for those two fields.  Turns out nobody was accessing them yet anyway.